presto-main/src/main/resources/webapp/assets/query.js

```diff
@@ -893,10 +893,30 @@ var QueryDetail = React.createClass({
     renderFailureInfo: function() {
         var query = this.state.query;
         if (query.failureInfo) {
+            var taskInfo = "N/A";
+            if (query.failureInfo.task != null) {
+                var task = query.failureInfo.task;
+                var tasks = this.getTasksFromStage(this.state.lastSnapshotTasks)
+                                .filter(function(task) { return this.state.taskFilter(task.taskStatus.state) }, this);
+                if(tasks && task.length != 0) {
```

Comment:
Make the 'tasks' check explicit. Javascript makes it very hard to confidently reason about what evaluates to true and false.